app/models/customer.rb

```diff
@@ -4,20 +4,32 @@
 #
 #  id              :integer          not null, primary key
 #  name            :string(255)      not null
-#  organization_id :integer
+#  organization_id :integer          not null
 #  created_at      :datetime
 #  updated_at      :datetime
+#  deleted_at      :datetime
 #
 
 class Customer < ActiveRecord::Base
+  acts_as_paranoid
+
   belongs_to :organization, inverse_of: :customers
   has_many :transactions, dependent: :destroy, inverse_of: :customer
 
-  validates :name, presence: true
   validates :organization, presence: true
-  validates :name, uniqueness: { scope: :organization_id }
+  validates :name, presence: true
+  validates :name, uniqueness: { scope: [:organization_id , :deleted_at] }
+
+  # gem 'paranoia' doesn't run validations on restore
+  before_restore :name_uniqueness
 
   def to_s
     name.truncate(30)
   end
+
+  private
+    def name_uniqueness
+      errors.add(:name, 'has been taken', strict: true) if Customer.find_by(
```

Comment:
@shir I dont quite understand. What do you mean? Can I somehow exmplicitly run validate_uniqueness_of? Didn't find something like that.. Can run record.validate_presense_of(:attr) etc, but seems to me there is no record.validate_uniqueness_of(:attr)